InnerEye/ML/deep_learning_config.py

```diff
@@ -283,6 +287,51 @@ def get_effective_random_seed(self) -> int:
             seed += self.cross_validation_split_index
         return seed
 
+    def get_hyperdrive_config(self, run_config: ScriptRunConfig) -> HyperDriveConfig:
+        """
+        Returns the HyperDrive config for either parameter search or cross validation
+        (if number_of_cross_validation_splits > 1).
+        :param run_config: AzureML estimator
+        :return: HyperDriveConfigs
+        """
+        if self.perform_cross_validation:
+            return self.get_cross_validation_hyperdrive_config(run_config)
+        else:
+            return self.get_parameter_search_hyperdrive_config(run_config)
+
+    def get_cross_validation_hyperdrive_sampler(self) -> GridParameterSampling:
+        """
+        Returns the cross validation sampler, required to sample the entire parameter space for cross validation.
+        """
+        return GridParameterSampling(parameter_space={
+            CROSS_VALIDATION_SPLIT_INDEX_TAG_KEY: choice(list(range(self.number_of_cross_validation_splits))),
+        })
+
+    @abstractmethod
+    def get_cross_validation_hyperdrive_config(self, run_config: ScriptRunConfig) -> HyperDriveConfig:
+        """
+        Returns a configuration for AzureML Hyperdrive that varies the cross validation split index.
+
+        This abstract method is implemented in both ModelConfigBase and LightningContainer
+        :param run_config: The AzureML run configuration object that configures training for an individual model.
+        :return: A hyperdrive configuration object.
+        """
+        pass
+
+    @abstractmethod
+    def get_parameter_search_hyperdrive_config(self, run_config: ScriptRunConfig) -> HyperDriveConfig:
+        """
+        Turns the given AzureML estimator (settings for running a job in AzureML) into a configuration object
+        for doing hyperparameter searches.
+
+        This abstract method is implemented in ModelConfigBase but not in LightningContainer. Classes inheriting
+        from LightningCountainer will wish to provide an implementation if they provide hyper drive support for
+        parameter search.
+        :param run_config: AzureML estimator
+        :return: HyperDriveConfigs
+        """
+        pass
```

Comment:
Thinking about that again... I think that's at the wrong place of the class hierarchy. It would be better if that lived at LightningContainer level. InnerEye container could implement that by falling back to model_config.get_parameter_search.... This would save you also one switch statement in your code (runner.py line 224)

Reply:
But it needs to be in a parent of both LightningContainer and DeepLEarningConfig. For example, it is overridden in InnerEye.configs.segmentation.HelleWorld which inherits from SegmentationModelBase -> ModelConfigBase -> DeepLearningConfig -> WorkflowParam. No?

Reply:
Done, just waiting for tests to run. I just moved `get_parameter_search_hyperdrive_config` back to `ModelConfigBase` which does not save me the switch you mention in runner, as that is about a different (but similarly named) method `get_hyperdrive_config`. Did you want all of them moved?

Reply:
Done for all the methods.